File: code/maps.cs
IReadOnlyDictionary&lt;string,string> occupations = 
                        new Dictionary&lt;string, string>
{
    ["Malcolm"] = "Captain",
    ["Kaylee"] = "Mechanic"
};

var occupationsMutable = new Dictionary&lt;string, string>
{
    ["Malcolm"] = "Captain",
    ["Kaylee"] = "Mechanic"
};
occupationsMutable["Jayne"] = "Public Relations";
occupationsMutable.Add("Rick", "Navigation");

File: code/maps.kt
val occupations = mapOf(
    "Malcolm" to "Captain",
    "Kaylee" to "Mechanic"
)

val occupationsMutable = mutableMapOf(
    "Malcolm" to "Captain",
    "Kaylee" to "Mechanic"
)
occupationsMutable["Jayne"] = "Public Relations"
occupationsMutable.put("Rick", "Navigation")
